### PR TITLE
chore: decouple editor components

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
Decouple editor components and codemirror from the main chunk, the main chunk has dropped by 200Kb to 1.1Mb

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11593047595>
> Commit: 4a5b3bcb8b56062c8c7feb711d7776bbae51bcbb
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11593047595&attempt=3" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 30 Oct 2024 13:57:44 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Implemented lazy loading for several components, enhancing performance and user experience during loading states.
	- Introduced new error handling UI with `EntityNotFoundPane` for better feedback when data is unavailable.
	- Added `LazilyLoadedChecklist` for optimized loading of the checklist component in onboarding.

- **Bug Fixes**
	- Improved asynchronous handling in test cases for more reliable rendering assertions.

- **Documentation**
	- Updated export statements for components to default exports for consistency.

- **Chores**
	- Refactored routing logic to utilize lazy-loaded components across various sections of the application.

### CODE REVIEW

File: app/client/src/pages/Editor/JSEditor/JSBlankState.tsx [hunk 1]
```diff
@@ -26,4 +26,4 @@ const JSBlankState = () => {
   );
 };
 
-export { JSBlankState };
+export default JSBlankState;
```
Comment (rajatagrawal) [inline app/client/src/pages/Editor/JSEditor/JSBlankState.tsx:29]: Isn't it preferable to avoid default exports ?

Comment (vsvamsi1) [reply app/client/src/pages/Editor/JSEditor/JSBlankState.tsx:29]: To import a component lazily i has to be a default export, that is the reason why i had to refactor it from a named export.

**Review by dvj1988 — APPROVED** [review]
Good job @vsvamsi1 👏 🚀 . It makes sense not to have editor code in the main bundle. 
LGTM

**Pull request merged**